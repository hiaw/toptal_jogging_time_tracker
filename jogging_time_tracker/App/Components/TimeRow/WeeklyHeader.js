/* @flow*/
import React from 'react'
import { Text, View } from 'react-native'
import moment from 'moment'

import styles from './Styles/WeeklyAverageRow.style.js'

export type Props = {
  date: string,
}

const WeeklyAverageRow = (props: Props) => {
  const { date } = props
  const dateText = `Week ${moment(date).isoWeek()}, ${moment(date).year()}`

  return (
    <View style={[styles.container, styles.header]}>
      <Text style={styles.welcome}>
        {dateText}
      </Text>
    </View>
  )
}

export default WeeklyAverageRow
