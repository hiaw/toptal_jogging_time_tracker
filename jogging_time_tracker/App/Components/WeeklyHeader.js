import React from 'react'
import { Text, View } from 'react-native'
import moment from 'moment'

import styles from './Styles/WeeklyAverageRow.style.js'

const WeeklyAverageRow = props => {
  const { date } = props
  const dateText = `Week ${moment(date).isoWeek()}, ${moment(date).year()}`

  return (
    <View style={[styles.container, styles.header]}>
      <Text style={styles.welcome}>
        {dateText}
      </Text>
    </View>
  )
}

export default WeeklyAverageRow
