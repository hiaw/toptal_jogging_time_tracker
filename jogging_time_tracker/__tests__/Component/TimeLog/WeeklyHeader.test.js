import 'react-native'
import React from 'react'

import WeeklyHeader from '../../../App/Components/TimeLog/WeeklyHeader.js'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

const noop = () => {}

it('renders correctly', () => {
  const tree = renderer
    .create(<WeeklyHeader date="2017-06-23T04:51:34.022Z" />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
