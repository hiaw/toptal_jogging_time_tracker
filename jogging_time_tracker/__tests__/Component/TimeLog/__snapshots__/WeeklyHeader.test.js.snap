// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "aliceblue",
        "borderBottomColor": "black",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingHorizontal": 20,
        "paddingVertical": 10,
      },
      Object {
        "backgroundColor": "cornsilk",
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Week 25, 2017
  </Text>
</View>
`;
